crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -161,6 +162,9 @@ impl CommandQueue {
 
 impl Drop for CommandQueue {
     fn drop(&mut self) {
+        if !self.bytes.is_empty() {
```

Comment:
I think we should only warn in debug mode and save an additional check here

Reply:
The bevy book [recommends compiling bevy with `opt-level=3`](https://bevyengine.org/learn/book/getting-started/setup/#compile-with-performance-optimizations). Disabling in debug mode means people who follow those instructions won't see the warning, making it pretty useless.

I do agree however that we could avoid checks, and additional binary bloat (log statements add a fairly large number of instructions, without counting the log message) But not sure how. I see some alternatives:

- Use `info!`, so that the log can be disabled at compile-time using the `log/max_level_warn` feature
- (controversial I think) use the `detailed_trace` feature
- Add a new feature, enabled by default, something like `costly_runtime_checks` and use it as a feature gate for the check.

Honestly, I don't prefer any in particular, all have their appeal and tradeoffs, and I don't think it's very important to avoid the check/log here.

Reply:
Something like `#[cfg(debug_assertions)]`?

Reply:
> The bevy book recommends compiling bevy with opt-level=3. Disabling in debug mode means people who follow those instructions won't see the warning, making it pretty useless.

Debug checks are still included when compiling with optimizations, as long as you don't switch to release mode. 

That said, I don't think it's worth spending too much thought on this check. For regular systems, or exclusive systems that properly reuse their SystemState, command queues will only get dropped when the app exit. The only time this check will be called unnecessarily is if users create a new SystemState every time the system runs, which is a performance pitfall of its own that's going to matter a lot more than this check.